### ISSUE TITLE
No module named 'splinter' error

### ISSUE THREAD

**Issue by vbaranov** [TokenMarketNet/ico#51]
```
Python 3.5.3
Parity 1.7
Kovan chain
```

All installation steps are passed from the [docs](https://ico.readthedocs.io/en/latest/install.html#setting-up-osx)

`deploy-token` generates `No module named 'splinter'` error after token contract deploying (but should note, that contract is created)

Full log:
```
(venv) viktors-MacBook-Pro:TokenMarketNet_ico viktor$ deploy-token --chain=kovan --contract-name="CentrallyIssuedToken" --address=0x005364854d51A0A12cb3cb9A402ef8b30702a565  --name="TestToken" --symbol=MOO --supply=916 --decimals=0 --verify --verify-filename=CentrallyIssuedToken.sol
Web3 provider is RPC connection http://localhost:8545
Deployer address is 0x005364854d51A0A12cb3cb9A402ef8b30702a565
Deployer balance is 0.985119332 ETH
Starting contract deployment
Contract address is 0xf4e3f2f959e1da62ef30ff0684ba88a7b0db4bad
CrowdsaleToken constructor arguments is 000000000000000000000000005364854d51a0a12cb3cb9a402ef8b30702a56500000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000e000000000000000000000000000000000000000000000000000000000000003940000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000954657374546f6b656e000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000034d4f4f0000000000000000000000000000000000000000000000000000000000
Traceback (most recent call last):
  File "/Users/viktor/Documents/HintsProjects/Test/TokenMarketNet_ico/ico/etherscan.py", line 35, in verify_contract
    from splinter import Browser
ImportError: No module named 'splinter'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/viktor/Documents/HintsProjects/Test/TokenMarketNet_ico/venv/bin/deploy-token", line 11, in <module>
    load_entry_point('ico', 'console_scripts', 'deploy-token')()
  File "/Users/viktor/Documents/HintsProjects/Test/TokenMarketNet_ico/venv/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/viktor/Documents/HintsProjects/Test/TokenMarketNet_ico/venv/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/viktor/Documents/HintsProjects/Test/TokenMarketNet_ico/venv/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/viktor/Documents/HintsProjects/Test/TokenMarketNet_ico/venv/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/viktor/Documents/HintsProjects/Test/TokenMarketNet_ico/ico/cmd/deploytoken.py", line 113, in main
    browser_driver=browser_driver)
  File "/Users/viktor/Documents/HintsProjects/Test/TokenMarketNet_ico/ico/etherscan.py", line 37, in verify_contract
    raise RuntimeError("Splinter package must be installed for verification automation")
RuntimeError: Splinter package must be installed for verification automation
```

**Comment by miohtama** [comment]
Thank you. This should be handled so that if `verify_contracts` is on in YAML, it refuses to run `deploy-contracts` if Splinter is not present.

Splinter is not included in the default installation, as it brings a lot of dependency packages that might be hard to manage.

The simple workaround is to do `pip install splinter` and re-run the deploy.

I'll leave this issue open, so that there is a better error handling in the future.

**Comment by vbaranov** [comment]
@miohtama tnx for quick response.
After installation of splinter `pip install splinter` as a workaround, I'm faced with this error:
```
Traceback (most recent call last):
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/lib/python3.5/site-packages/selenium/webdriver/common/service.py", line 74, in start
    stdout=self.log_file, stderr=self.log_file)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/subprocess.py", line 676, in __init__
    restore_signals, start_new_session)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/subprocess.py", line 1282, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: 'chromedriver'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/bin/deploy-token", line 11, in <module>
    load_entry_point('ico', 'console_scripts', 'deploy-token')()
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/viktor/Documents/HintsProjects/Test/ico/ico/cmd/deploytoken.py", line 113, in main
    browser_driver=browser_driver)
  File "/Users/viktor/Documents/HintsProjects/Test/ico/ico/etherscan.py", line 56, in verify_contract
    with Browser(driver_name=browser_driver) as browser:
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/lib/python3.5/site-packages/splinter/browser.py", line 63, in Browser
    return driver(*args, **kwargs)
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/lib/python3.5/site-packages/splinter/driver/webdriver/chrome.py", line 34, in __init__
    self.driver = Chrome(chrome_options=options, **kwargs)
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/lib/python3.5/site-packages/selenium/webdriver/chrome/webdriver.py", line 62, in __init__
    self.service.start()
  File "/Users/viktor/Documents/HintsProjects/Test/ico/venv/lib/python3.5/site-packages/selenium/webdriver/common/service.py", line 81, in start
    os.path.basename(self.path), self.start_error_message)
selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home
```
Am I missing something else?

**Comment by miohtama** [comment]
Ah.

`chromedriver` is a special service used to automate Chrome browser for Selenium automation.

As you are on OSX and (hopefully) using Homebrew try these instructions:

http://brewformulas.org/Chromedriver

**Comment by rstormsf** [comment]
@miohtama I installed `chromedriver` 
```
 chromedriver
Starting ChromeDriver 2.31.488774 (7e15618d1bf16df8bf0ecf2914ed1964a387ba0b) on port 9515
Only local connections are allowed.
```


```
Waiting EtherScan to process the contract verification
Contract verification failed. Check the browser for details.
Press enter to continue
Traceback (most recent call last):
  File "/Users/rstorm/repos/tokenmarket/pop_init/ico/venv/bin/deploy-contracts", line 11, in <module>
    load_entry_point('ico', 'console_scripts', 'deploy-contracts')()
  File "/Users/rstorm/repos/tokenmarket/pop_init/ico/venv/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/rstorm/repos/tokenmarket/pop_init/ico/venv/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/rstorm/repos/tokenmarket/pop_init/ico/venv/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/rstorm/repos/tokenmarket/pop_init/ico/venv/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/rstorm/repos/tokenmarket/pop_init/ico/ico/cmd/deploycontracts.py", line 31, in main
    deploy_crowdsale_from_file(project, deployment_file, deployment_name, address)
  File "/Users/rstorm/repos/tokenmarket/pop_init/ico/ico/deploy.py", line 265, in deploy_crowdsale_from_file
    return _deploy_contracts(project, chain, web3, yaml_filename, chain_data, deploy_address)
  File "/Users/rstorm/repos/tokenmarket/pop_init/ico/ico/deploy.py", line 249, in _deploy_contracts
    runtime_data, statistics, contracts = deploy_crowdsale(project, chain, chain_data, deploy_address)
  File "/Users/rstorm/repos/tokenmarket/pop_init/ico/ico/deploy.py", line 151, in deploy_crowdsale
    with open(expanded_path, "wt") as out:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/rstorm/repos/tokenmarket/pop_init/ico/build/expanded/CrowdsaleToken.sol'
```

looks like Selenium didn't start the process as well. I've faced issues like that before with incompatible versions of chromedriver. I'm curious what version are you running?

**Comment by rstormsf** [comment]
FYI -
```
Expanding source code file CrowdsaleToken.sol
Expanding source code file ./StandardToken.sol
Expanding source code file zeppelin/contracts/token/ERC20.sol
Expanding source code file ./ERC20Basic.sol
Expanding source code file ./SafeMath.sol
Expanding source code file ./UpgradeableToken.sol
Expanding source code file ./UpgradeAgent.sol
Expanding source code file ./ReleasableToken.sol
Expanding source code file zeppelin/contracts/ownership/Ownable.sol
Expanding source code file ./MintableToken.sol
Expanding source code file ./SafeMathLib.sol
```

```
 tree build
build
└── contracts.json

0 directories, 1 file
```
there is only 1 file in `build` folder.... another reason it failed

**Comment by miohtama** [comment]
Let me get this fixed.

**Comment by rstormsf** [comment]
There is also hard-coded compiler version:
https://github.com/TokenMarketNet/ico/blob/master/ico/etherscan.py#L24

**Comment by miohtama** [comment]
See @55c64d87cab132fe05f9d7cf4fa0dcb1673b6219

**Comment by miohtama** [comment]
Ok - can you open another issue on the compiler version - I'll get it fixed tomorrow.

**Comment by miohtama** [comment]
We were only 0.4.8 compatible until very recently (few days ago).

**Comment by rstormsf** [comment]
@miohtama created issue #52

**Comment by rstormsf** [comment]
https://github.com/TokenMarketNet/ico/commit/55c64d87cab132fe05f9d7cf4fa0dcb1673b6219 
it only changes the README file ?  @miohtama

**Comment by miohtama** [comment]
Fixed in db37eaa2c31d0387a022e3b2dd69fe9223d24fdf

**Comment by Fonger** [comment]
I think this should be documented
